### DISCUSSION TITLE
Let's discuss the Utility system.

### DISCUSSION THREAD

**Discussion by SniperGoth** [redeclipse/base#1241] [Ideas and Feature Requests]
This is a mechanic that has been talked about for more than half a decade at this point, and was never really put too much thought into.

Utilities would be like items that you could use to help you, or your team, on a match, maybe on a cooldown, or  replenished by a special energy pick up.

The only one we have that is *kinda* confirmed that will make an appearance, is the Grappling Gun, which im the person that is taking care of making the model currently.

![grappling gun](https://user-images.githubusercontent.com/24359960/109200182-0eee9200-777f-11eb-9ec7-aea3374d8d29.png)

But, what about the others? What would they be?

Maybe a adrenaline shot that makes you go 25% faster for 10 seconds? A portable shield that you can place? A gun turret? Or even a portable lift/pusher?

Since 2.1 is still very far away from releasing, we might as well consider putting at least some utilities to begin with.

Any ideas?

**Comment by dc-redeclipse** [comment]
Many ideas... maybe to many...
here are some:

- different turret models depending on the weapon type:
 - gun turret model only on type smg, pistol, rifle, shotgun 
 - rocket launcher turret model only on weapon type rocket
 - flamer turret model only on weapon type flamer
- possibility to backflip and /180 turn on while wallclimb
- triggered rails
- WebRTC for in-game and out- to in-game communications connections (to replace irc)
- NPCs (kill able pedestrians idea)
- single and multiplayer missions
- matchmaking system / kinda player lobby
- server site executable scripts in maps (maybe an flag on which instance the script will be executed)
- python <-> cube script wrapper
- ranking system to motivate player for playing RE more often maybe in combinations with achievements like cool vanities only with specific ranks
- server flag to store costume maps uploaded by players and make them select able
- new game mode mutant: 30% player vs 70% players only with claws slower less life points catch the player turn them to mutant and win after mutating all players, players win only after surviving the hole match at least only one
- gravity material, material with other configurable gravity perfect for parkour
- slow motion power item for parkour
- power item for parkour
- knief as an alternative for kick for near fights but on same player position level
... and many more

**Reply by SniperGoth** [reply]
The only issue is that these have nothing to do with what utility items/abilities are lmao, but they would fit more of a wishlist of sorts i'd say.

**Comment by Jigoku** [comment]
I wonder if the torch vanity thing that was discussed before could fit into that, as some kind of equipped item.

**Comment by dc-redeclipse** [comment]
Next attempt after sleep:
- turret item:
  gun turret model only on type smg, pistol, rifle, shotgun with a lifetime limit
  rocket launcher turret model only on weapon type rocket with a lifetime limit
  flamer turret model only on weapon type flamer with a lifetime limit
- barrier item:
  a drop able zapper wall barrier by 2 x 4 meters per element with a lifetime limit, they are connectable by dropping an other on it left or right will expand on the position
- roller item:
  drop able with a lifetime limit no team damage
- heal station item:
  drop a container that can healt player for a limited time

**Comment by ghost** [comment]
I suggest add jetpacks.

**Reply by SniperGoth** [reply]
We had jetpacks before as a mutator, but it was kinda useless tbh, because you moved the same speed as running.
I mean, i wouldn't be against the idea of bringing it back as a utility item, but it would probably need to be reworked to become useful, maybe giving it a better boost, or it only lifting you upwards, who knows.

**Comment by dc-redeclipse** [comment]
stealth mode pickup - makes the player invisible for a limited time if its activated.

**Comment by janitor-raus** [comment]
I don't know about utilities as actual handhelds, but you can definitely have powerups (or powerdowns) to add some variety. Some ideas off the top of my head:

Suction Boots -- Free directional wallrunning without falling off unless you jump off yourself, but it increases the amount of damage you can take during wallruns, especially from fire (spiders anyone). Weapon recoil and bullet spread is increased.

Heavy Armor -- Increases your damage resistance a certain amount and prevents critical hits, but it similarly limits your movement speed and wallruns to half, and you're unable to impulse jump. Player size is increase, thus making you a slightly larger target.

Stealth Camo -- Turns you invisible for a short amount of time, but you're unable to attack for the duration nor impulse jump. Comes with a free mine that you can set down, but only in its cloaked state.

Infinity Clip -- Increases your weapon damage, and allows you to fire non-stop for the duration. On the other hand, you can no longer move when you start shooting, your recoil is increased, and your firing rate is slowed down, weapon precision increased.

Homing Fire -- Projectiles (bullets, grenades, rockets) get a weak homing capability towards the enemy closest to them, acting as aim assist, but their damage is reduced by half. Projectile speed is also slightly reduced, and their critical damage slightly increased.

Human Torch -- For ~10 seconds, be engulfed by flames and become invincible, unable to wield weapons to attack aside from punching. A flame field will be deployed for the duration at your location, dealing the burning debuff to all players in its radious at 4 times the normal damage rate once inflicted. Movement speed is increased by 50%. If the time runs out or if you fall in water, you will automatically die. Being attacked with a flamer will add some time to your timer. Your next spawn (only) will be capped at 50% health and 75% movement speed.

Grenadier -- Your weapons (and any you pick up) are replaced with grenades. Left mouse button will throw a sticky proximity taser-nade that deals 20% of typical damage and stuns targets for 2 seconds. Right mouse button with throw a bouncy cluster-nade that will pop into 15 smaller bomblets when near a player or after 3 bounces, dealing 5% of a typical grenade's damage in a small radious. Grenades thrown toward your close proximity or mines (visible or not) are automatically defused and replenish your own stock if lacking to shoot back. Explosion damage (rockets) is nullified entirely. Can only return to a normal state after respawning.
